docs/apis/plugintypes/theme/index.md

```diff
@@ -0,0 +1,466 @@
+---
+title: Theme plugins
+tags:
+  - Plugins
+  - Theme
+---
+
+A Moodle theme allows users to customize the appearance and functionality of their Moodle site, from overall design to specific activities. Users can create their own themes or modify existing ones, leveraging CSS and JavaScript for customization. The theme architecture ensures smooth fallbacks for minimal changes, fostering flexibility and ease of use.
+
+## File structure
+
+import {
+    Lang,
+    Lib,
+    VersionPHP,
+} from '../../_files';
+
+Theme plugins are located in the `/theme` directory.
+
+Each plugin is in a separate subdirectory and consists of a number of _mandatory files_ and any other files the developer is going to use.
+
+<details>
+  <summary>View an example directory layout for a `theme_example` plugin.</summary>
+
+```console
+ theme/example
+ |-- amd
+ |   └-- src
+ |-- classes
+ |   └-- output
+ |-- fonts
+ |-- fonts_core
+ |-- fonts_plugins
+ |   └-- plugintype
+ |        └-- pluginname
+ |-- lang
+ |   └-- en
+ |       └-- theme_example.php
+ |-- layout
+ |-- less
+ |-- pix
+ |   └-- favicon.ico
+ |   └-- screenshot.png
+ |-- pix_plugins
+ |   └-- plugintype
+ |        └-- pluginname
+ |-- style
+ |-- scss
+ |-- templates
+ |-- config.php
+ |-- settings.php
+ └-- version.php
+```
+
+</details>
+
+Some of the important files for the Theme plugintype are described below. See the [common plugin files](../commonfiles) documentation for details of other files which may be useful in your plugin.
+
+:::tip Override icons and templates
+
+You can customize icons in base themes without altering core code by placing them in `dataroot/pix` and `dataroot/pix_plugins`. If a theme extends a base theme and includes its own icons, those will take precedence.
+
+Similarly, mustache templates in base themes can be overridden without impacting core code by placing them in `templates/[componentname]/[templatename].mustache`.
```

Comment:
But this is where? Is this relative to dataroot? Or your own theme?

Reply:
I've improved this sections adding the alternative (which is overriding these files in the theme).